reference-project/src/main/java/com/devonfw/demoquarkus/rest/v1/controller/SecureRestController.java

```diff
@@ -0,0 +1,85 @@
+package com.devonfw.demoquarkus.rest.v1.controller;
+
+import org.eclipse.microprofile.jwt.JsonWebToken;
+import javax.annotation.security.PermitAll;
+import javax.annotation.security.RolesAllowed;
+import javax.enterprise.context.RequestScoped;
+import javax.inject.Inject;
+import javax.ws.rs.*;
+import javax.ws.rs.core.*;
+
+@RequestScoped
+@Path("/secure")
+@Produces(MediaType.APPLICATION_JSON)
+@Consumes(MediaType.APPLICATION_JSON)
+public class SecureRestController {
+    
+    @Inject
+    JsonWebToken jwt;
+    
+    @GET()
+    @Path("/permit")
+    @PermitAll 
+    @Produces(MediaType.TEXT_PLAIN)
+    public String permit() {
+        return "Access to /secure/permit was granted!";
+    }
+    
+    @GET()
+    @Path("/token")
+    @PermitAll
+    @Produces(MediaType.TEXT_PLAIN)
+    public String token(@Context SecurityContext ctx) {
+        String name;
+        if (ctx.getUserPrincipal() == null) { 
+            name = "anonymous";
+        } else if (!ctx.getUserPrincipal().getName().equals(jwt.getName())) { 
+            throw new InternalServerErrorException("Principal and JsonWebToken names do not match");
```

Comment:
it is always good to assert things but is there any real reason how this could ever happen?
Sounds like inconsistency in quarkus itself if this case ever occurs...